homeassistant/components/light/__init__.py

```diff
@@ -374,9 +374,11 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:  # noqa:
     profiles = hass.data[DATA_PROFILES] = Profiles(hass)
     await profiles.async_initialize()
 
-    def preprocess_data(data: dict[str, Any]) -> dict[str, Any]:
+    def preprocess_data(
+        data: dict[str | vol.Optional, Any]
```

Comment:
Let's leave `data` as `dict[str, Any]`. That's basically how we typed all vol schema validation functions so far.

If at some point annotations are added to voluptuous, we'll need to go over them again anyway.

--
`base` is different as it's constructed with keys from `cv.ENTITY_SERVICE_FIELDS`.

Reply:
But by having data as `dict[str, Any]` we end up `entity_field: Never`.

Reply:
> But by having data as `dict[str, Any]` we end up `entity_field: Never`.

I just took another look. `dict[str, Any]` is indeed correct for `data`. The resulting `entity_field: Never` is technically correct. However, the `__hash__` implementation from `vol.Optional` is just `hash(self.schema)`. I.e. for simple strings the hash of the string. That's why the dict contains check works, even though the types don't actually overlap.